### PR TITLE
[#5540] Improvement(dev/release): Update the `LICENSE` file to fix the wrong path problem

### PR DESCRIPTION
**Pull request by liuchunhao**
### What changes were proposed in this pull request?

  	This script checks if all the files mentioned in the LICENSE file are present in the project.
  	Besides highlighting the missing files, it will display possible locations for any new files.
  	This script adds text like [NOT FOUND] --> "new/path/to/the/file" in the end of the line where the file is missing.
  	With the output of this script, you can update the LICENSE file with the correct file paths.

        new file: 
            ./dev/release/check-license.sh   (script for checking the missing files)

        modified:
            ./LICENSE    (remove missing files or replace them with the newly found files)

### Why are the changes needed?

	Some code paths listed in LICENSE file are already outdated due to the refactoring of the codes.
    We need to update the path to reflect the actual one.

### Does this PR introduce _any_ user-facing change?
	No

### How was this patch tested?
```bash
./check-license.sh
```
This script adds [NOT FOUND] --> "new/path/to/the/file" in the end of the line where the file is missing.

replace with newly found files

This script will show the following message if all checks pass:

all good

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -296,15 +295,14 @@
...
-   ./security-admin/src/main/java/org/apache/ranger/view/VXUser.java
-   ./security-admin/src/main/java/org/apache/ranger/view/VXUserList.java
+   ./authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/reference/JsonDateSerializer.java
+   ./authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/reference/RangerCommonEnums.java
+   ./authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/reference/VList.java
+   ./authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/reference/VXDataObject.java
+   ./authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/reference/VXGroup.java
+   ./authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/reference/VXGroupList.java
+   ./authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/reference/VXUser.java
+   ./authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/reference/VXUserList.java
```
Comment (jerryshao) [inline LICENSE:308]: Hi @liuchunhao thanks for your work. Can you please move this Ranger part under "Apache Arrow" and above "MIT License"? Thanks.

File: dev/release/check-license.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#
+
+# This script checks if all the files mentioned in the LICENSE file are present in the project.
+# Besides highlighting the missing files, it will display possible locations for any new files.
+# It adds text like [NOT FOUND] --> "new/path/to/the/file" in the end of the line where the file is missing.
+# With the output of this script, you can update the LICENSE file with the correct file paths.
+
+FAILED=0
+
+PROJECT_ROOT=$(dirname $(dirname $(dirname $(readlink -f "$0"))))
```
Comment (jerryshao) [inline dev/release/check-license.sh:27]: Can you please follow the `update-java-doc-version.sh` to let user can also specify the project dir, like:

```
PROJECT_DIR=${2:-$(cd "$(dirname "$0")" && pwd)/../../}
```

Comment (jerryshao) [reply dev/release/check-license.sh:27]: Besides, did you verify both on Linux and macOS. My concern is that some commands may not be compatible between Linux and Mac.

File: LICENSE (hunk 1 above)
Comment (liuchunhao) [reply LICENSE:302]: Sure, I've adjusted it as requested.

File: dev/release/check-license.sh (hunk 2 above)
Comment (liuchunhao) [reply dev/release/check-license.sh]: Hi,  I've adjusted as requested and ensure the update works on both Mac and Linux. 
I think the "#!/usr/bin/env bash" can provide the best compatibility across different systems

Below is the test result on ubuntu 22.04
Screenshot 2024-11-13 at 1 43 44 PM

**Review by jerryshao — APPROVED** [review]
LGTM, thanks a lot for the fix.

**Pull request merged**